File: app/src/main/java/life/plank/juna/zone/view/adapter/user/GetCoinsAdapter.java
package life.plank.juna.zone.view.adapter.user;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import life.plank.juna.zone.R;
import life.plank.juna.zone.data.model.CoinPack;
import life.plank.juna.zone.util.BaseRecyclerView;
import life.plank.juna.zone.util.ObliqueStrikeTextView;

public class GetCoinsAdapter extends BaseRecyclerView.Adapter<GetCoinsAdapter.GetCoinsViewHolder> {

    private List<CoinPack> coinPackList;

    public GetCoinsAdapter() {
        this.coinPackList = new ArrayList<>();
    }

    @Override
    public GetCoinsViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new GetCoinsViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_get_coins, parent, false));
    }

    @Override
    public void onBindViewHolder(GetCoinsViewHolder holder, int position) {
        holder.bind(coinPackList.get(position));
    }

    @Override
    public int getItemCount() {
        return 0;
    }

    /**
     * Use this method to update the recyclerView's contents instead of using and managing a list in the Activity (where it doesn't really belong).
     *
     * @param coinPackList the list to update (usually fetched from the server).
     */
    public void update(List<CoinPack> coinPackList) {
        this.coinPackList = coinPackList;
        notifyDataSetChanged();
    }

    static class GetCoinsViewHolder extends BaseRecyclerView.ViewHolder {

        @BindView(R.id.coin_amount)
        TextView coinAmount;
        @BindView(R.id.earlier_price)
        ObliqueStrikeTextView earlierPrice;
        @BindView(R.id.current_price)
        TextView currentPrice;

        GetCoinsViewHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        public void bind(CoinPack coinPack) {
            if (coinPack.getCoinCount() > 0) coinAmount.setText(coinPack.getCoinCount());
            if (coinPack.getEarlierPrice() > 0)
                earlierPrice.setText(String.valueOf(coinPack.getEarlierPrice()));
            if (coinPack.getCurrentPrice() > 0)
                currentPrice.setText(String.valueOf(coinPack.getCurrentPrice()));
        }
    }
}

File: app/src/main/java/life/plank/juna/zone/view/user/adapter/GetCoinsAdapter.kt
package life.plank.juna.zone.view.user.adapter

import android.view.*
import androidx.recyclerview.widget.RecyclerView
import kotlinx.android.synthetic.main.item_get_coins.view.*
import life.plank.juna.zone.R
import life.plank.juna.zone.data.model.user.CoinPack
import java.util.*

class GetCoinsAdapter : RecyclerView.Adapter<GetCoinsAdapter.GetCoinsViewHolder>() {

    private var coinPackList: List<CoinPack> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GetCoinsViewHolder =
            GetCoinsViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_get_coins, parent, false))

    override fun onBindViewHolder(holder: GetCoinsViewHolder, position: Int) {
        holder.itemView.run {
            val coinPack = coinPackList[position]
            if (coinPack.coinCount > 0) coin_amount.setText(coinPack.coinCount)
            if (coinPack.earlierPrice > 0)
                earlier_price.text = coinPack.earlierPrice.toString()
            if (coinPack.currentPrice > 0)
                current_price.text = coinPack.currentPrice.toString()
        }
    }

    override fun getItemCount(): Int = coinPackList.size

    fun update(coinPackList: List<CoinPack>) {
        this.coinPackList = coinPackList
        notifyDataSetChanged()
    }

    class GetCoinsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}